### DISCUSSION TITLE
Home and Play Views

### DISCUSSION THREAD

**Discussion by kevincoleman** [online-go/online-go.com#2366] [Ideas]
Our discussion about the quickmatch finder has brought up something I’ve not been able to put my finger on for a bit: the functions and goals of the Home vs Play pages.

Several times recently I’ve found myself confused on which to click. If I want to play my currently active games I go to Play, right? Wrong. If I go to home I expect to see a general overview of what’s going on on OGS, right? Wrong.

Then enter our discussion about where a user should go to start a game, and it’s got the wheels turning.

Before I get any crazy ideas, can anyone provide clarity on what the goals are of the Play and Home pages?

The goal of the Home page is _____
The goal of the Play page is _____

It could just be a naming thing, or it could be a lack of direction—or heck, it could just be me. Thoughts appreciated!

**Comment by anoek** [comment]
Hearing you describe it, it's perhaps odd. Here's how we got to where we are:

Every site needs a home page right? Some starting place. Especially in the beginning, a very large fraction of our player base is correspondence players, and so it seemed natural that when you landed on the site the first thing you'd want to see was all of the games you were playing so you can make your moves. So that's the "Home" screen. Could be a better name would fit better, but then again people are used to the concept and name of "Home screen", as that's common vernacular so something needs to fill that spot.

Of course the other very common thing people want to do is start a new game, and all of the match making stuff, which started off with a seek graph system, which required a lot of real estate to make any sense of, needed it's own page. It probably would have been better named "Matchmaking" or something, but I went with "Play", maybe because it was shorter in length, or maybe just because that's what came to mind first and 10 years later it's still just kinda whatever I named it. 

So that's the history of the nomenclature, and in short the functionality boils down to the need to display lots of ongoing games, and the need to to display lots of open games, and the separation of pages simply boils down to my inability to reconcile an interface that can do both of those things in a clear and clean way in one page.

**Comment by kevincoleman** [comment]
Ah! Thanks for the history of it. It does make more sense. I wonder if we could revisit the way it’s presented to help solve any confusion and/or to streamline the game creation process.

This is still just a vague idea, and I know it’s a heavily trafficked menu area, so we would want to consider all angles and test, but what if we made the Play page linked from a new “Start a Game” (blue?) button in the nav? That way it would feel more like an action item and less like a page to visit. This could be a pretty minor iteration at first if we had it replace the “Play” nav item. It would make it very apparent to new players that they can start a game quickly from anywhere.

**Reply by anoek** [reply]
I my minds eye it looks weird to have a button up there, but it might not be hard to beat my minds eye in creating something that looks good.

If we did do that, it seems to me the right thing to do would be to open up a matchmaking modal of some sort, as opposed to navigating away from wherever they are to the Play page, would you agree?

**Reply by kevincoleman** [reply]
Yeah it does seem different, for sure. That said, it being different could be the thing that calls enough attention to it so people know it’s not just a nav item.

Here’s an example of a header with a button in it from a site I did some work on. Seemed to work surprisingly well (if you don’t count the yellow for bad contrast):

![Screenshot 2023-09-18 at 7 43 19 AM](https://github.com/online-go/online-go.com/assets/445707/3ff02a97-0d73-4145-a261-611c3fae6d53)
(Edit: they actually still used it as a menu item with a dropdown, which I found odd, but the visual treatment is what I was after here).

I’m currently a bit torn between using a modal for new games and using the page. In a full desktop experience I think a modal might be good, but on mobile a modal for that much content doesn’t really make sense, and we’d just lose some margin on the side for the visuals of it.

**Reply by kevincoleman** [reply]
So if we were to add a button we’d need to think about its function as a menu item. Currently there are three areas to the Play section (Play, Tournament, Ladders) and that provides some great functionality. I’m not sure I’d want to reorganize such a key menu area willy-nilly, but maybe we could get away with some renaming? To me the most natural descriptions of these areas are:

“My Games” (or just “Games”)

and

“Start a New Game”

This leads me to the next conundrum: Neither the Tournament page nor the Ladders page is about starting games. Would we want to consider moving those elsewhere?

I think for a menu change of this high of traffic we may want to consider a “what’s changed” modal for existing players. Not sure on this, though. If the menu self-describes it may not be necessary. Thoughts?

**Reply by anoek** [reply]
I'm hesitant to drop in a name change to "Start a New Game" for two reasons: 1) While I acknowledge that we have folks bringing up an issue that it's not fast enough / intuitive enough to start a new game, it's not clear to me that this would "fix" that. We already have a single blue button on the home page that brings folks to the place where they start a new game, so if that's not enough for them it's not clear to me that a link on the nav bar would fix that either. 2) It's long, less is more in the nav bar I feel like. Especially when we start translating to some of the more lengthy languages like German.

My Games (or Games) is an easier sell. We should probably rename the Watch > Games link too if we were to do that though.

**Reply by kevincoleman** [reply]
Yeah I agree that “Start a New Game” is long and could be more succinct. Just trying to think of how to make that action (starting a new game) dead simple and natural, site wide. Maybe just “New Game” or even a button with “+ Game” could help keep it short if we decided we wanted to do this.

<img width="1413" alt="Screenshot 2023-09-19 at 1 20 09 PM" src="https://github.com/online-go/online-go.com/assets/445707/82979df1-06db-49d0-97d4-31eb5f6b45a0">

A sample of what it might feel like to have a button up there. (Note: “Compete” was added to house Tournaments and Ladders)

> While I acknowledge that we have folks bringing up an issue that it's not fast enough / intuitive enough to start a new game, it's not clear to me that this would "fix" that.

Without stats and tests I can’t really posit that anything will be clear to everyone, myself included. That said, here are some reasons I can think of in favor of the button in the bar:
1. It’s always in the same place. No hunting to find where to make a new game, ever.
2. It doesn’t hide itself based on your view’s state (like when you’re in a single game view or a non-empty home view).
3. It’d be highlighted in a block of color, making it even easier to find.
4. It’d be natural to think of it as an action with an outcome, as opposed to a place you navigate to.
5. Being right next to the logo there would be _no question_ for new players how to get started.

Perhaps it could go without saying, but if we tried this and found people had a harder time creating games (again, stats) we could revert. This could be experimental and or A/B tested.

> We already have a single blue button on the home page

Very true for blitz/live players, but not for correspondence players since they often have several games going.

One other idea that I keep coming back to since you mentioned it: make the whole new-game experience a modal, always, and have users stay on their home page. Earlier I thought this may be non-optimal on mobile, but if we just remove some padding/margins it’s actually the same as having a dedicated page. This modal idea jives with making a button for New Game very nicely. In a sense, this is almost what we have.

One of the key things that implicates the use of a modal is that the content of it is temporary. Secondarily there’s usually a user interaction that’s single-goal focused. Both of those things are basically true of the Play page. You go there only for a little bit while you find a new game, and the goal is to accept/create a new game.

I don’t know if we have stats to validate this, but I suspect the majority of traffic coming _to_ the Play page is from a finished game (especially for live players) or from the home page (more for correspondence players). So the flow looks like this:

Game end (or caught up on correspondence games) > Play Page > Get a new game > Play new game

The more I look at it, the more natural it feels.

Just an aside: I’ve got some thoughts on the menu in general, which I made a new discussion for [here](https://github.com/online-go/online-go.com/discussions/2371).

**Reply by anoek** [reply]
As an initial impression kind of thing I think there's certainly some appeal to that look, however I do have some concern about the boldness of the button up there. I worry we'll get a bit of push back from the community with a bright blue button present in their games as it calls attention to it instead of the game their playing. I suppose one solution would be to set it to a more subtle gray color on game pages when you're one of the players, or something of the sort.

If we do end up with a button up there then I think the expectation would be that it would open a modal, rather than simply redirecting to the Play page or something. 

I think it's certainly worth keeping in mind, but I  would want to ensure it fit in with whatever holistic vision we settle on before committing to anything.

**Reply by kevincoleman** [reply]
I’m not sure I totally agree that it would distract from a game, as they’re in different areas and with different contexts, but I’m happy to differ on that. There’s plenty of room for a solution in between. Take for example this still-buttony button that’s not as big a block of color. This does reduce the ease of finding the button, but only marginally.

![Screenshot 2023-09-20 at 12 00 48 PM](https://github.com/online-go/online-go.com/assets/445707/dfb4ef23-b03c-4c2f-84ec-0668704c89f9)

I tend to agree that we should try this button out with the modal.

Is this something we should try drafting up in dev/experimental? I’d be happy to try to make an initial PR for it.

**Reply by anoek** [reply]
Sure we can try it out on experimental. Do you have any thoughts on the modal?

**Reply by kevincoleman** [reply]
I think the most natural way to iterate would be to just use the whole content of the Play page in a modal. If we find that’s too clunky or something then we could try problem solving from there.

![Screenshot 2023-09-21 at 11 18 58 AM](https://github.com/online-go/online-go.com/assets/445707/5a84bcd2-dc5f-4465-979b-bbaaf0a0de45)

Huh. I think I see our first problem, though. Anything from that page that launched a modal now launches a second modal... hmm.

**Reply by shinuito** [reply]
> Huh. I think I see our first problem, though. Anything from that page that launched a modal now launches a second modal... hmm.

Yeah generally I think less clicks will be appreciated in order to get a game over modal-ception.

One thing that really holds us back a bit in design I think, compared to lichess or chesscom is that we have so many time settings options. I think it's great to have these because I've played both fischer and byo-yomi games for instance in leagues on OGS, but also even canadian at one point for an online blitz tournament. So offering the options is great, but it does mean either one needs a lot of automatch button options (which may mean more queues), or one has this thing that we have now, where no-one really knows the time settings of the game going into the game, only that it's ~30s per move ish, or ~10s etc, or really how long the game might take.

Previous suggestions had essentially suggested kind of combining the home and play pages in some way, either you have tabs to swap between showing your own games, and showing automatch buttons or challenges or squash a bunch of things in.

Clossius put a nice bit of effort on some ideas in case you were interested in looking at them for inspiration or critique @kevincoleman 

https://forums.online-go.com/t/new-landing-page-idea/46491?u=shinuito
